lib/cc/cli/source_fingerprint.rb

```diff
@@ -0,0 +1,69 @@
+require "digest/md5"
+
+module CC
+  module CLI
+    class SourceFingerprint
+      def initialize(output)
+        @output = output
+      end
+
+      def compute
+        md5 = Digest::MD5.new
+        md5 << key
+        md5.hexdigest
+      end
+
+      def key
+        key = "#{path}|#{check_name}"
+        key << "|#{source.gsub(/\s+/, "")}" if source
```

Comment:
This is where I'm concerned about using the `|` separators - it could lead to collisions here (maybe thats ok)